### PR TITLE
Freshen up deps

### PR DESCRIPTION
**Pull request by bcomnes**
Looking to close https://github.com/hypermodules/gh-release/issues/92.  Starting with all the outdated deps to get those out of the way.

The deps that need code changes appear to be:

```
√ gh-release % npm outdated
Package  Current  Wanted  Latest  Location
ghauth     3.2.1   3.2.1   4.0.0  gh-release
shelljs    0.3.0   0.3.0   0.8.4  gh-release
yargs      2.3.0   2.3.0  15.4.1  gh-release
```

Working on that part next.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 language: node_js
 node_js:
-  - 4
-  - 6
-  - 'stable'
+  - 10
+  - 12
```
Comment (jgravois) [inline .travis.yml:4]: for projects that i rarely touch, i've grown pretty fond of just having Travis just run the tests on whatever versions happen to be current at runtime.

```suggestion
  - 'lts/*'
  - 'node'
```
https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions

if you prefer to be explicit, no offense taken.

Comment (bcomnes) [reply .travis.yml:4]: Thats what I like to do as well, but I was trying to match the pattern that was present.  Happy to switch to that if @ungoldman is cool with it.

**Comment by bcomnes** [comment]
Ok, deps are all up to date.  This first and foremost will remove the security warnings.  

Unfortunately, ghauth doesn't have a fix for the new endpoint yet, but I opened an upstream issue: https://github.com/rvagg/ghauth/issues/26

I would hold off from releasing this (we can land the PR though) until I figure out a solution for the auth flow.  It would be a nice contribution to ghuath to update it to the latest auth endpoint.

File: .travis.yml (hunk 1 above)
Comment (ungoldman) [reply .travis.yml:4]: fine by me

**Comment by ungoldman** [comment]
If this isn't a breaking change why not release now?

**Comment by bcomnes** [comment]
A bunch of deps raised the minimum node version required to run, so from that perspective its breaking, but if we already assumed LTS was always the minimum it wouldn't be.  wdyt?

Breaking would be safer but more downstream work.

**Comment by bcomnes** [comment]
The new auth flow would likely be breaking, so I figured I would roll all this in with that.

**Comment by ungoldman** [comment]
ok makes sense to me. just like to avoid leaving things unreleased in master when possible -- we're all busy and it's entirely possible the new auth flow won't get dealt with soon

**Comment by bcomnes** [comment]
Sounds good. If the auth stuff stalls out, I can cut a release.  Whats your preference semverwise on just this portion of the work?

**Comment by ungoldman** [comment]
considering this gets 16k downloads / month it would probably be better to be cautious and cut a major release -- I've been bitten enough by other module authors releasing breaking changes under a patch or minor to try to never do that to anyone else.

we _could_ wait to merge this until the auth flow stuff is also ready to not do two major releases

**Comment by bcomnes** [comment]
Ok sounds good.  Ill made a call to merge over the next week or so depending on upstream interest in the new device auth flow.

**Comment by ungoldman** [comment]
thanks bret, sorry to complicate it. appreciate the work

**Comment by paulcpederson** [comment]
Doing the good lordt's work here, @bcomnes much appreciated 👍

**Comment by bcomnes** [comment]
Pr opened: https://github.com/rvagg/ghauth/pull/27

Any reviews appreciated.  Biggest change that affects gh-release is the change from `authUrl` to `githubHost`.

```
- the `options.authUrl` (default: `https://api.github.com/authorizations`) is removed in favor of `options.githubHost` (default `github.com`).
```

The reason being, there are a ton more URLs and subdomains involved now, so providing the host seems like the best thing to customize.

**Comment by bcomnes** [comment]
Looks like we've stalled out upstream.  Thinking now that we should land this work, and then just break again when upstream gh auth is out.

**Comment by ungoldman** [comment]
@bcomnes we could fork ghauth until upstream is responsive -- not sure if rvagg is actively maintaining ghauth anymore

**Comment by bcomnes** [comment]
Definitely an option I was thinking about.  Would be nice to at least get his input on the desired outcome.  I'll give it a couple of weeks.

**Comment by bcomnes** [comment]
Ok, I forked ghauth for the time being, while we wait for upstream.  Going to test this out a few more times and cut a major release.

**Comment by bcomnes** [comment]
Found a bug where the error messaging is terrible when you try to release to an org that is lacking permissions for the new oauth scoped tokens.  Going to improve that.

**Comment by bcomnes** [comment]
Ok, all good.  Can't really tell the difference between a 404 or a bad org permissions setting, so I'll just rely on the context of where the error happens. Will improve if needed.

Screen Shot 2020-09-13 at 10 05 48 AM

**Comment by bcomnes** [comment]
OK, going to cut a major on this.

**Comment by bcomnes** [comment]
Ok this is out in `4.0.0`

**Comment by paulcpederson** [comment]
Thanks @bcomnes !

**Pull request merged**